.github/workflows/build.yml

```diff
@@ -443,12 +443,104 @@ jobs:
           files: |
             rpa-worker-e2e-tests/build/test-results/**/*.xml
   
+  macos_sign_and_notarize:
```

Comment:
How come the Mac signing is in this new job, rather than being at the end of the jobs that build and zip the binary, like the Windows one does?

Reply:
This would allow us to still build artifacts even if signing fails. Do you think this is useful? If so, I can adjust our windows pipeline to reflect that as well